### PR TITLE
Improve the hibernation doc

### LINKED ISSUE #5606: Shoot cluster is hibernated even the hibernation is configured to false
**How to categorize this issue?**

/area shoot cluster
/area hibernation
/area usability
/kind bug

**What happened**: When you want to deploy shoot cluster without hibernation  and you specify hibernation part in shoot cluster definition yaml with shceduling.

```
hibernation:
    enabled: false
    schedules:
    - location: Europe/Bratislava
      start: 00 17 * * 1,2,3,4,5
```
hibernation is disabled only if this part is deleted from yaml or if schedules part is removed.

```
hibernation:
    enabled: false
```
On that case hibernation enabled value is ignored and shoot cluster will be hibernated anyway.  

**What you expected to happen**:
When I specify hibernation  to enabled: false it should not check schedules and should disable hibernation.  

**How to reproduce it (as minimally and precisely as possible)**:
create yaml manifest for shoot cluster where you specify hibernation part following:
```
hibernation:
    enabled: false
    schedules:
    - location: Europe/Bratislava
      start: 00 17 * * 1,2,3,4,5
```

**Anything else we need to know?**:

**Environment**:

- Gardener version: 1.43
- Kubernetes version (use `kubectl version`): 1.21.X
- Cloud provider or hardware configuration: Openstack

### PR DESCRIPTION
**Pull request by ialidzhikov**
/area documentation

Fixes https://github.com/gardener/gardener/issues/5606

**Release note**:

```other operator
NONE
```

### CODE REVIEW

File: docs/usage/shoot_hibernate.md [hunk 1]
```diff
@@ -35,6 +35,8 @@ To scale up everything where it was before hibernation, Gardener doesn’t delet
 
 ## Hibernate your cluster manually
 
+The `.spec.hibernation.enabled` field specifies whether the cluster needs to be hibernated or not. If the field is set to `true`, the cluster's desired state is to be hibernated. If it is set to `false` or not specified at all, the cluster's desired state is to be awaken.
```
Comment (n-boshnakov) [inline docs/usage/shoot_hibernate.md]:
```suggestion
The `.spec.hibernation.enabled` field specifies whether the cluster needs to be hibernated or not. If the field is set to `true`, the cluster's desired state is to be hibernated. If it is set to `false` or not specified at all, the cluster's desired state is to be awakened.
```

**Review by n-boshnakov — CHANGES_REQUESTED** [review]

**Review by n-boshnakov — APPROVED** [review]
lgtm

**Review by rfranzke — APPROVED** [review]
/lgtm
/milestone v1.43

**Pull request merged**